### ISSUE TITLE
How do we reduce overfitting

### ISSUE THREAD

**Issue by ShreePoudel0** [biomodhub/biomod2#620]
I am having serious overfitting in my models. How do we reduce it. I have tried many different settings and parameters but the result is same,
This is the code I use

myBiomodData <- 
  BIOMOD_FormatingData(           #Gathers all input data necessary to run biomod2 models.
    resp.var = data["Conflict"], #Response Variable ['Column containing presence points']
    resp.xy = data[, c('X', 'Y')], #Response Variable ['Column containing Longitude and Latitude']
    expl.var = myExpl, #Predictor Variable
    resp.name = "Conflict", #Name of the response variable
    PA.nb.rep = 3, #No. of replication for presence data
    PA.nb.absences = 100, #No. of absence points to generate or background points
    PA.strategy = 'random',#Strategy for selecting background points
    #filter.raster = TRUE
  )

myRFoptions <- list(nodesize = 20, maxnodes = 10, mtry = 2)
user.RF <- rep( list(myRFoptions), (ncol(myBiomodData@PA.table) + 1 ))
names(user.RF) <- c( paste0("_", names(myBiomodData@PA.table), "_allRun"), "_allData_allRun")

user.GBM <- list('_allData_allRun' = list(n.trees = 500, 
                                          interaction.depth = 1,
                                          n.minobsinnode = 20,
                                          shrinkage = 0.001,
                                          cv.folds = 10))

user.val <- list(RF.binary.randomForest.randomForest = user.RF,
                 GBM.binary.gbm.gbm = user.GBM)

myOpt <- bm_ModelingOptions(bm.format = myBiomodData,
                            data.type = 'binary',
                            models = c('GBM', 'RF','GLM','GAM','MAXENT'),
                            strategy = "user.defined",
                            user.val = user.val,
                            user.base = "bigboss")

myBiomodModelOut <- BIOMOD_Modeling(bm.format = myBiomodData, 
                                    models = c('RF'),
                                    var.import = 1,
                                    CV.strategy = 'random',
                                    CV.nb.rep = 10,
                                    CV.k = 5,
                                    CV.perc = 0.7, 
                                    OPT.strategy = 'user.defined',
                                    OPT.user = myOpt,
                                    prevalence = 0.5, 
                                    metric.eval = c('TSS', 'ROC'))

bm_PlotEvalMean(myBiomodModelOut)
bm_PlotEvalBoxplot(myBiomodModelOut,
                   group.by = c("algo", "run"),
                   do.plot = TRUE)

full.name      PA    run algo
1        Conflict_PA1_RUN1_RF     PA1   RUN1   RF
2        Conflict_PA1_RUN1_RF     PA1   RUN1   RF
3        Conflict_PA1_RUN2_RF     PA1   RUN2   RF
4        Conflict_PA1_RUN2_RF     PA1   RUN2   RF
5        Conflict_PA1_RUN3_RF     PA1   RUN3   RF
6        Conflict_PA1_RUN3_RF     PA1   RUN3   RF
7        Conflict_PA1_RUN4_RF     PA1   RUN4   RF
8        Conflict_PA1_RUN4_RF     PA1   RUN4   RF
9        Conflict_PA1_RUN5_RF     PA1   RUN5   RF
10       Conflict_PA1_RUN5_RF     PA1   RUN5   RF
11       Conflict_PA1_RUN6_RF     PA1   RUN6   RF
12       Conflict_PA1_RUN6_RF     PA1   RUN6   RF
13       Conflict_PA1_RUN7_RF     PA1   RUN7   RF
14       Conflict_PA1_RUN7_RF     PA1   RUN7   RF
15       Conflict_PA1_RUN8_RF     PA1   RUN8   RF
16       Conflict_PA1_RUN8_RF     PA1   RUN8   RF
17       Conflict_PA1_RUN9_RF     PA1   RUN9   RF
18       Conflict_PA1_RUN9_RF     PA1   RUN9   RF
19      Conflict_PA1_RUN10_RF     PA1  RUN10   RF
20      Conflict_PA1_RUN10_RF     PA1  RUN10   RF
21     Conflict_PA1_allRun_RF     PA1 allRun   RF
22     Conflict_PA1_allRun_RF     PA1 allRun   RF
23       Conflict_PA2_RUN1_RF     PA2   RUN1   RF
24       Conflict_PA2_RUN1_RF     PA2   RUN1   RF
25       Conflict_PA2_RUN2_RF     PA2   RUN2   RF
26       Conflict_PA2_RUN2_RF     PA2   RUN2   RF
27       Conflict_PA2_RUN3_RF     PA2   RUN3   RF
28       Conflict_PA2_RUN3_RF     PA2   RUN3   RF
29       Conflict_PA2_RUN4_RF     PA2   RUN4   RF
30       Conflict_PA2_RUN4_RF     PA2   RUN4   RF
31       Conflict_PA2_RUN5_RF     PA2   RUN5   RF
32       Conflict_PA2_RUN5_RF     PA2   RUN5   RF
33       Conflict_PA2_RUN6_RF     PA2   RUN6   RF
34       Conflict_PA2_RUN6_RF     PA2   RUN6   RF
35       Conflict_PA2_RUN7_RF     PA2   RUN7   RF
36       Conflict_PA2_RUN7_RF     PA2   RUN7   RF
37       Conflict_PA2_RUN8_RF     PA2   RUN8   RF
38       Conflict_PA2_RUN8_RF     PA2   RUN8   RF
39       Conflict_PA2_RUN9_RF     PA2   RUN9   RF
40       Conflict_PA2_RUN9_RF     PA2   RUN9   RF
41      Conflict_PA2_RUN10_RF     PA2  RUN10   RF
42      Conflict_PA2_RUN10_RF     PA2  RUN10   RF
43     Conflict_PA2_allRun_RF     PA2 allRun   RF
44     Conflict_PA2_allRun_RF     PA2 allRun   RF
45       Conflict_PA3_RUN1_RF     PA3   RUN1   RF
46       Conflict_PA3_RUN1_RF     PA3   RUN1   RF
47       Conflict_PA3_RUN2_RF     PA3   RUN2   RF
48       Conflict_PA3_RUN2_RF     PA3   RUN2   RF
49       Conflict_PA3_RUN3_RF     PA3   RUN3   RF
50       Conflict_PA3_RUN3_RF     PA3   RUN3   RF
51       Conflict_PA3_RUN4_RF     PA3   RUN4   RF
52       Conflict_PA3_RUN4_RF     PA3   RUN4   RF
53       Conflict_PA3_RUN5_RF     PA3   RUN5   RF
54       Conflict_PA3_RUN5_RF     PA3   RUN5   RF
55       Conflict_PA3_RUN6_RF     PA3   RUN6   RF
56       Conflict_PA3_RUN6_RF     PA3   RUN6   RF
57       Conflict_PA3_RUN7_RF     PA3   RUN7   RF
58       Conflict_PA3_RUN7_RF     PA3   RUN7   RF
59       Conflict_PA3_RUN8_RF     PA3   RUN8   RF
60       Conflict_PA3_RUN8_RF     PA3   RUN8   RF
61       Conflict_PA3_RUN9_RF     PA3   RUN9   RF
62       Conflict_PA3_RUN9_RF     PA3   RUN9   RF
63      Conflict_PA3_RUN10_RF     PA3  RUN10   RF
64      Conflict_PA3_RUN10_RF     PA3  RUN10   RF
65     Conflict_PA3_allRun_RF     PA3 allRun   RF
66     Conflict_PA3_allRun_RF     PA3 allRun   RF
67 Conflict_allData_allRun_RF allData allRun   RF
68 Conflict_allData_allRun_RF allData allRun   RF
   metric.eval cutoff sensitivity specificity
1          TSS  380.0      86.667      82.857
2          ROC  379.0      86.667      82.857
3          TSS  441.0      83.333      85.714
4          ROC  442.0      83.333      85.714
5          TSS  346.0      91.667      78.571
6          ROC  350.0      91.667      80.000
7          TSS  377.0      86.667      81.429
8          ROC  339.0      90.000      78.571
9          TSS  430.0      88.333      87.143
10         ROC  431.0      88.333      87.143
11         TSS  345.0      93.333      80.000
12         ROC  346.0      93.333      80.000
13         TSS  476.0      81.667      91.429
14         ROC  474.0      81.667      91.429
15         TSS  487.0      90.000      92.857
16         ROC  490.0      90.000      92.857
17         TSS  417.0      91.667      87.143
18         ROC  418.0      91.667      87.143
19         TSS  469.0      81.667      88.571
20         ROC  470.0      81.667      88.571
21         TSS  432.0      87.209      83.000
22         ROC  458.0      86.047      85.000
23         TSS  385.5      91.667      77.143
24         ROC  384.0      91.667      77.143
25         TSS  484.0      88.333      84.286
26         ROC  485.0      88.333      84.286
27         TSS  537.0      80.000      85.714
28         ROC  366.0      90.000      75.714
29         TSS  536.0      81.667      91.429
30         ROC  540.0      81.667      91.429
31         TSS  331.0      91.667      81.429
32         ROC  333.0      91.667      81.429
33         TSS  320.0      93.333      74.286
34         ROC  321.0      93.333      74.286
35         TSS  506.0      86.667      87.143
36         ROC  507.0      86.667      87.143
37         TSS  586.0      73.333      97.143
38         ROC  586.0      73.333      97.143
39         TSS  433.0      86.667      87.143
40         ROC  435.0      86.667      87.143
41         TSS  462.0      86.667      87.143
42         ROC  463.0      86.667      87.143
43         TSS  521.0      84.884      86.000
44         ROC  507.0      87.209      84.000
45         TSS  477.0      80.000      82.857
46         ROC  485.0      80.000      84.286
47         TSS  410.0      85.000      82.857
48         ROC  416.0      85.000      84.286
49         TSS  466.0      83.333      84.286
50         ROC  467.0      83.333      85.714
51         TSS  408.0      90.000      82.857
52         ROC  401.0      91.667      82.857
53         TSS  515.0      83.333      84.286
54         ROC  511.0      83.333      84.286
55         TSS  435.0      83.333      85.714
56         ROC  435.0      83.333      85.714
57         TSS  403.0      86.667      84.286
58         ROC  403.0      86.667      84.286
59         TSS  419.0      85.000      84.286
60         ROC  417.0      85.000      84.286
61         TSS  427.0      83.333      87.143
62         ROC  429.0      83.333      87.143
63         TSS  467.0      88.333      82.857
64         ROC  470.0      88.333      84.286
65         TSS  468.0      82.558      88.000
66         ROC  468.0      82.558      88.000
67         TSS  155.0      89.535      82.000
68         ROC  157.0      89.535      82.333
   calibration validation evaluation
1        0.695      0.405         NA
2        0.925      0.783         NA
3        0.690      0.415         NA
4        0.925      0.828         NA
5        0.717      0.441         NA
6        0.925      0.808         NA
7        0.681      0.490         NA
8        0.918      0.863         NA
9        0.755      0.569         NA
10       0.935      0.858         NA
11       0.733      0.469         NA
12       0.928      0.792         NA
13       0.731      0.372         NA
14       0.936      0.773         NA
15       0.829      0.172         NA
16       0.943      0.642         NA
17       0.788      0.159         NA
18       0.946      0.697         NA
19       0.702      0.608         NA
20       0.919      0.891         NA
21       0.702         NA         NA
22       0.928         NA         NA
23       0.688      0.690         NA
24       0.915      0.862         NA
25       0.726      0.579         NA
26       0.930      0.831         NA
27       0.657      0.492         NA
28       0.918      0.842         NA
29       0.731      0.536         NA
30       0.921      0.860         NA
31       0.731      0.397         NA
32       0.938      0.722         NA
33       0.676      0.656         NA
34       0.915      0.831         NA
35       0.738      0.354         NA
36       0.936      0.735         NA
37       0.705      0.338         NA
38       0.929      0.845         NA
39       0.738      0.233         NA
40       0.925      0.770         NA
41       0.738      0.569         NA
42       0.934      0.799         NA
43       0.709         NA         NA
44       0.930         NA         NA
45       0.629      0.292         NA
46       0.910      0.784         NA
47       0.679      0.441         NA
48       0.915      0.730         NA
49       0.690      0.349         NA
50       0.916      0.753         NA
51       0.729      0.254         NA
52       0.920      0.754         NA
53       0.676      0.426         NA
54       0.905      0.812         NA
55       0.690      0.277         NA
56       0.919      0.684         NA
57       0.710      0.195         NA
58       0.925      0.597         NA
59       0.693      0.144         NA
60       0.920      0.737         NA
61       0.705      0.349         NA
62       0.923      0.790         NA
63       0.712      0.210         NA
64       0.919      0.694         NA
65       0.706         NA         NA
66       0.914         NA         NA
67       0.715         NA         NA
68       0.919         NA         NA

The calibration and validation are very different. and also in suitability map, it is showing highly suitable where there is no presence points and there shouldn't be any suitability. For now, I am doing RF only to see results fast but GBM also shows overfitting
Thank you I really need help
